flytekit/image_spec/image_spec.py

```diff
@@ -278,6 +278,16 @@ def exist(self) -> Optional[bool]:
                     f"    pip install --upgrade docker"
                 )
 
+            if isinstance(e, DockerException) and "Error while fetching server API version" in str(e):
```

Comment:
nit: I'd like to see the original error still.
```suggestion
            click.secho(f"Failed to check if the image exists with error:\n {e}", fg="red")
            if isinstance(e, DockerException) and "Error while fetching server API version" in str(e):
```

Reply:
Thanks! I'll make sure the original error is still shown.